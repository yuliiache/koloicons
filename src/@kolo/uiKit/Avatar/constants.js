export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
