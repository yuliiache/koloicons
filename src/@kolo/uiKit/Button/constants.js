export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
