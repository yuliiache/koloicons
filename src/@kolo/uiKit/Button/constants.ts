export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
