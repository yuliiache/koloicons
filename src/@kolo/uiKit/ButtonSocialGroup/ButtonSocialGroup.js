import React from 'react';

import styles from './ButtonSocialGroup.module.scss';

export const ButtonSocialGroup = ({children}) => {
  return (
    <ul className={styles.buttonSocialGroup}>
      {React.Children.map(children, (child) => (
        <li className={styles.buttonSocialGroupItem}>{child}</li>
      ))}
    </ul>
  );
};
