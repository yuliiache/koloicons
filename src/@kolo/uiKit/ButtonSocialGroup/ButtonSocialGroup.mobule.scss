.buttonSocialGroup {
  display: flex;
  width: 100%;
  margin: 0;
  padding: 0;

  list-style: none;

  &Item {
    flex-basis: 33.333%;
    margin-right: 2px;
    overflow: hidden;

    &:first-of-type {
      border-top-left-radius: 4px;
      border-bottom-left-radius: 4px;
    }

    &:last-of-type {
      margin-right: 0;
    }

    &:last-of-type {
      border-top-right-radius: 4px;
      border-bottom-right-radius: 4px;
    }
  }
}
