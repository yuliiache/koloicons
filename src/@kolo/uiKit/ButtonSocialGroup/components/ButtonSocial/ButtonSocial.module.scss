@import "@kolo/design";

.buttonSocial {
  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  padding: 12px 16px;

  font-family: $mainFontFamily;
  font-weight: 500;
  font-size: 0.7rem;
  text-align: center;
  color: $secFontColor;

  background-color: $WhiteBG;
  border: none;
  cursor: pointer;

  transition: box-shadow 0.2s ease-out;

  &:hover {
    color: $BlueColor;
  }

  &:focus {
    position: relative;
    outline: 4px solid rgba($color: $BlueColor, $alpha: 0.2);
    z-index: 1;
  }

  &Icon {
    display: inline-flex;

    & svg {
      width: 24px;
      height: 24px;
    }
  }

  &Text {
    margin-right: 16px;
  }

  &Disconnected {
    &:hover {
      color: $RedColor;

      &Text,
      &Icon {
        opacity: 1;
      }
    }

    &Text,
    &Icon {
      opacity: 0.7;
    }
  }
}
