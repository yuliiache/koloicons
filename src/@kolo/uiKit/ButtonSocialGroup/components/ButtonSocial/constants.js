export const BUTTON_SOCIAL_TEST_ID = 'buttonSocialComponent';

export const BUTTON_SOCIAL_TEXT_TEST_ID = `${BUTTON_SOCIAL_TEST_ID}Text`;
