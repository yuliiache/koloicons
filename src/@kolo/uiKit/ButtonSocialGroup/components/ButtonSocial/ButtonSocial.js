import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import {BUTTON_SOCIAL_TEST_ID, BUTTON_SOCIAL_TEXT_TEST_ID} from './constants';

import styles from './ButtonSocial.module.scss';

export const ButtonSocial = ({text, shouldShowStatus, isConnected, renderIcon, onClick}) => {
  const isDisconnected = shouldShowStatus && !isConnected;

  const buttonClasses = classNames(styles.buttonSocial, {
    [styles.buttonSocialDisconnected]: isDisconnected,
  });

  let buttonText = text;

  if (shouldShowStatus) {
    buttonText = isConnected ? `Connect ${text}` : `Disconnect ${text}`;
  }

  return (
    <button
      data-testid={BUTTON_SOCIAL_TEST_ID}
      className={classNames(buttonClasses)}
      type="button"
      onClick={onClick}
    >
      <span
        className={styles.buttonSocialText}
        data-testid={BUTTON_SOCIAL_TEXT_TEST_ID}
      >
        {buttonText}
      </span>
      <i className={styles.buttonSocialIcon}>{renderIcon && renderIcon()}</i>
    </button>
  );
};

ButtonSocial.propTypes = {
  text: PropTypes.string,
  shouldShowStatus: PropTypes.bool,
  isConnected: PropTypes.bool,
  renderIcon: PropTypes.func,
  onClick: PropTypes.func,
};
