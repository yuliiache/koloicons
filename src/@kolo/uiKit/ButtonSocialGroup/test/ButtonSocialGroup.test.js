import React from 'react';
import {cleanup, render, screen} from '@testing-library/react';

import {ButtonSocialGroup} from '../ButtonSocialGroup';
import {ButtonSocial} from '../components/ButtonSocial/ButtonSocial';
import {BUTTON_SOCIAL_TEST_ID} from '../components/ButtonSocial/constants';

describe('ButtonSocialGroup component', () => {
  afterEach(() => cleanup());

  it('should render snapshot successfuly', () => {
    const props = {
      text: 'Facebook',
      renderIcon: () => <span>MockedIcon</span>,
      onClick: jest.fn(),
    };

    const {asFragment} = render(
      <ButtonSocialGroup>
        <ButtonSocial {...props} />
        <ButtonSocial
          shouldShowStatus={Boolean(true)}
          {...props}
        />
        <ButtonSocial
          shouldShowStatus={Boolean(true)}
          isConnected={Boolean(true)}
          {...props}
        />
      </ButtonSocialGroup>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render all button when button are provided', () => {
    const props = {
      text: 'Facebook',
      renderIcon: () => <span>MockedIcon</span>,
      onClick: jest.fn(),
    };

    render(
      <ButtonSocialGroup>
        <ButtonSocial {...props} />
        <ButtonSocial
          shouldShowStatus={Boolean(true)}
          {...props}
        />
        <ButtonSocial
          shouldShowStatus={Boolean(true)}
          isConnected={Boolean(true)}
          {...props}
        />
      </ButtonSocialGroup>
    );

    const buttonSocialElement = screen.getAllByTestId(BUTTON_SOCIAL_TEST_ID);

    expect(buttonSocialElement).toHaveLength(3);
  });
});
