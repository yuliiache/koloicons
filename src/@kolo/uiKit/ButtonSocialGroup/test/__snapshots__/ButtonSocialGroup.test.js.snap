// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSocialGroup component should render snapshot successfuly 1`] = `
<DocumentFragment>
  <ul
    class="buttonSocialGroup"
  >
    <li
      class="buttonSocialGroupItem"
    >
      <button
        class="buttonSocial"
        data-testid="buttonSocialComponent"
        type="button"
      >
        <span
          class="buttonSocialText"
          data-testid="buttonSocialComponentText"
        >
          Facebook
        </span>
        <i
          class="buttonSocialIcon"
        >
          <span>
            MockedIcon
          </span>
        </i>
      </button>
    </li>
    <li
      class="buttonSocialGroupItem"
    >
      <button
        class="buttonSocial buttonSocialDisconnected"
        data-testid="buttonSocialComponent"
        type="button"
      >
        <span
          class="buttonSocialText"
          data-testid="buttonSocialComponentText"
        >
          Disconnect Facebook
        </span>
        <i
          class="buttonSocialIcon"
        >
          <span>
            MockedIcon
          </span>
        </i>
      </button>
    </li>
    <li
      class="buttonSocialGroupItem"
    >
      <button
        class="buttonSocial"
        data-testid="buttonSocialComponent"
        type="button"
      >
        <span
          class="buttonSocialText"
          data-testid="buttonSocialComponentText"
        >
          Connect Facebook
        </span>
        <i
          class="buttonSocialIcon"
        >
          <span>
            MockedIcon
          </span>
        </i>
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
